src/Text/Pandoc/Shared.hs

```diff
@@ -630,6 +631,11 @@ compactify'DL items =
              | otherwise           -> items
            _                       -> items
 
+-- | Combine a list of lines by adding hard linebreaks.  This is useful
+--   e.g. for rudimentary support of LineBlock elements in writers.
+combineLines :: [[Inline]] -> [Inline]
+combineLines = concat . map (++ [LineBreak])
```

Comment:
My initial version did use `intercalate`, I do not quite remember why I changed it.  I think it was because rendering results were off in some formats (HTML), while it doesn't do any harm in other formats.  The switch in methods was my reason to introduce this function, exactly for the case where we'd want to change it again.

My current thoughts on it are that I'd rather add a function `linesToPara :: [[Inline]] -> Block` and export that instead.  This would save a tiny bit of boilerplate in the writers, as `combineLines` is almost exclusively used in combination with `Para`. Don't know why I didn't do this already.